internal/utils/client.go

```diff
@@ -40,18 +41,26 @@ func New(ctx context.Context, name, password, token, url, httpUserAgent string)
 	}
 }
 
-func (c *Client) UpdateCloudInitFile(cloudInitYAML string, key string, values []interface{}) (string, error) {
```

Comment:
i have mixed feelings about this function
1. single_value is a "boolean trap"
2. behaviour is a "magic string"
3. if single_value is true, and values is empty, this can cause a panic 

I think we can use smaller , single responsibility funcs instead